src/components/chip/chip.ts

```diff
@@ -81,7 +82,7 @@ export default class IgcChipComponent extends EventEmitterMixin<
    * @attr
    */
   @property({ reflect: true })
-  public variant!: 'primary' | 'success' | 'danger' | 'warning' | 'info';
+  public variant!: Variant;
```

Comment:
Oh cool, one more thing I noticed, technically this is the other way around: `variant?` instead of `variant!`.
Either that, or the chip has an extra state the badge does not - default , which isn't any of the variants.